src/main/java/org/opensearch/knn/indices/Model.java

```diff
@@ -190,7 +172,11 @@ private void writeOptionalModelBlob(StreamOutput output) throws IOException {
     public void writeTo(StreamOutput output) throws IOException {
         getModelMetadata().writeTo(output);
         writeOptionalModelBlob(output);
-        output.writeOptionalString(modelID);
+        output.writeString(modelID);
```

Comment:
Will there be problem if we write ModelId first instead of last?

Reply:
Any optional writes should be guarded by booleans whether they are written. So I dont see this as being a problem.Optional should be guarded by

Reply:
Shouldnt be. Optional reads/writes should be guarded by a boolean. I see that is done for writeOptionalModelBlob.